File: fixtures/fixture.js
a=1;
b=2;
c=-1;
d=[1,2];

// Ordering

assert(1 << 2 == 4);
assert((0+1)<<(0+2)==4);
assert(1<<2==4);
assert(a<<b==4);

assert(16 >> 2 == 4);
assert((8+8)>>(1+1)== 4);
assert(16>>2==4);
assert(b>>a==1);

assert(1 <= 2);
assert((0+1)<=(1+1));
assert(1<=2);
assert(a<=b);

assert(2 >= 1);
assert((1+1)>=(0+1));
assert(2>=1);
assert(b>=a);

assert(-1 >>> 1 == 2147483647);
assert((2-1)>>>(0+1)==2147483647);
assert(-1>>>1== 2147483647);
assert(c>>>a==2147483647);

// Equality

assert(1 != 2);
assert(1!=2);
assert((0+1)!=(1+1));
assert(a!=b);

assert(1 == 1);
assert(1==1);
assert((1+1)==(1+1));
assert(a==a);

assert(1 === 1);
assert(1===1);
assert((1+1)===(1+1));
assert(a===a);

// Logic

assert(!false);
assert(!(false));
assert(!0);

assert(true && true);
assert(true&&true);
assert(a&&b);
assert((true)&&(true));

assert(false || true);
assert(false||true);
assert((false)||(true));
assert(a||b);

// Sets

assert(1 in [1,2]);
assert((1)in([1,2]));

// Subscripts

a(= [0,1,2,3,4]);
assert(a[0]==0);
assert(a[1]==1);
assert(a[2]==2);
assert(a[3]==3);
assert(a[4]==4);

// Null

assert(null == null);
assert((null)==(null));
assert(null==null);

// Arithmetic

assert(Math.sqrt 16 == 4);

a = 0

assert(++a == 1);
assert((++a)==2);

assert(a++ == 2);
assert(a++==3);
assert((a++)==4);

// Undefined

assert(undefined == undefined);
assert((undefined)==(undefined));
assert(undefined==undefined);

assert(undefined == void 0);
assert((undefined)==(void 0));
assert(undefined==void 0);

File: fixtures/fixture.rb
#!/usr/bin/env ruby

require 'test/unit'

# punctuation

class Main < Test::Unit::TestCase
  def test_punctuation
    a = 1
    b = 10
    [1 .. 10]
    [1 ... 10]
    [1..10]
    [1...10]
    [(1)..(10)]
    [(1)...(10)]
    [a..b]
    [a...b]
  end

  def test_main
    a=1
    b=2

    # bitshift

    assert_equal 1 << 2, 4
    assert_equal (1)<<(2), 4
    assert_equal 1<<2, 4

    assert_equal 4 >> 1, 2
    assert_equal (4)>>(1), 2
    assert_equal 4>>1, 2

    # ordering

    assert 2 >= 1
    assert 2>=1
    assert (2)>=(1)
    assert b>=a

    assert 1 <= 2
    assert 1<=2
    assert (1)<=(2)
    assert a<=b

    # equality

    assert 1 == 1
    assert 1==1
    assert (1)==(1)
    assert a==a

    assert 1 != 2
    assert 1!=2
    assert (1)!=(2)
    assert a!=b

    assert Fixnum === 1
    assert Fixnum===1
    assert (Fixnum)===(1)
    assert Fixnum===a

    assert 'abc' =~ /abc/
    assert 'abc'=~/abc/
    assert ('abc')=~(/abc/)

    assert 'abc' !~ /cba/
    assert 'abc'!~/cba/
    assert ('abc')!~(/cba/)

    # logic

    # assert(not false)
    assert(not(false))

    assert ! false
    assert !false
    assert !(false)

    assert true and true
    assert (true)and(true)

    assert((false or true))
    assert(((false)or(true)))

    assert true && true
    assert true&&true
    assert (true)&&(true)

    assert false || true
    assert false||true
    assert (false)||(true)

    # arrows

    assert({:a => true}[:a])
    assert({:a=>true}[:a])
    assert({(:a)=>(true)}[:a])

    f = -> a { a**2 }
    f=->a{ a**2 }
    f = ->(a) { a**2 }
    assert_equal f.(3), 9

    # function

    f = lambda {|a| a**2}
    f=lambda{|a| a**2}
    assert_equal f.(4), 16

    # nil

    assert_nil nil
    assert_nil (nil)

  end
end
